### ISSUE TITLE
UX issues from Bug 1488654

### ISSUE THREAD

**Issue by juliandescottes** [devtools-html/debugger.html#6948]
From https://bugzilla.mozilla.org/show_bug.cgi?id=1488654
Forwarding from bugzilla, feel free to split more.

- [x] show indicator when pretty printing is in progress
  This can take a long time on big files, we could add some user feedback 
- [ ] highlight is too subtle
  When we jump to the debugger, the line only flashes for a brief moment
- [x] always reveal in tree 
  Synchronize the tree with the selected tab (I think we had this at some point?)
- [ ] max-width of editor tabs
  This would avoid situations such as shown below where 50% of the tabbar is empty and everything is always in the overlow menu
<img width="419" alt="image" src="https://user-images.githubusercontent.com/1141550/45205152-a1360f00-b281-11e8-9785-26060bc250f9.png">
  
@violasong @fvsch not sure if this is the best repository to discuss this or if it should move to the new UX issues repo?

**Comment by fvsch** [comment]
Highlight colors discussion is [devtools-html/ux#14](https://github.com/devtools-html/ux/issues/14)

**Comment by violasong** [comment]
Oh haha, markdown fail. Thanks!

**Comment by birtles** [comment]
>     * highlight too brief - interesting, I wonder if the highlight should continue persisting until closing the tab or if that would feel 'dirty'

I was actually thinking of something like Notepad++ where the current line (i.e. line where the cursor is) is highlighted. If you select multiple lines, the line where the selection's "focus" is, is highlighted.

**Comment by fvsch** [comment]
VS Code has something like that too. Depending on the theme it can look like a background color or an outline.

<img width="949" alt="screen shot" src="https://user-images.githubusercontent.com/243601/45545980-2ee2a300-b81c-11e8-99f5-99fd6080c0ac.png">

**Comment by violasong** [comment]
I just realized Style Editor also has a highlighted cursor line in a stronger shade than the Inspector hover color - I'd recommend trying these both out to see what feels more correct.

**Comment by akashs09** [comment]
/claim  
- max-width of editor tabs
-  highlight is too subtle

**Comment by juliandescottes** [comment]
If you feel like the current max-width is good compromise, feel free to ignore the last point. 

The issue is highlighted in the screenshot, it is pretty easy to only have one visible tab and everything in the overflow menu. To address if we could:
- reduce the max width
- adapt the max width to the available space

@birtles @violasong  any other idea?

**Comment by violasong** [comment]
Re: max-width of editor tabs - hard to say how this should work without playing around with it in code but I guess Firefox's tab behavior would be my inspiration :)

**Comment by jasonLaster** [comment]
closing this as it is hard to follow. If we want to follow up on the highlight line being too subtle or max width, lets file a new issue.
